lib/modules/dosomething/dosomething_signup/includes/Signup.php

```diff
@@ -89,14 +89,21 @@ private function build($data) {
     $this->id = $data->sid;
     $this->created_at = $data->timestamp;
 
-    $user = user_load($data->uid);
+    $drupal_user = user_load($data->uid);
+
+    $northstar_response = dosomething_northstar_get_northstar_user($data->uid);
+    $northstar_response = json_decode($northstar_response);
+
+    if (!empty($northstar_response->data) && !isset($northstar_response->error)) {
+      $northstar_user = $northstar_response->data;
+    }
 
     $this->user = [
-      'id' => $data->field_northstar_id_value === 'NONE' ? NULL : $data->field_northstar_id_value,
-      'first_name' => dosomething_helpers_extract_field_data($user->field_first_name),
-      'last_initial' => dosomething_helpers_extract_field_data($user->field_last_name),
-      'photo' => $user->picture,
-      'country' => dosomething_helpers_extract_field_data($user->field_address),
+      'id' => isset($northstar_user) ? $northstar_user->id : NULL,
+      'first_name' => dosomething_helpers_extract_field_data($drupal_user->field_first_name),
```

Comment:
These should be coming from the Northstar response, [like in the Reportback](https://github.com/DoSomething/phoenix/blob/728b9b8184f69fcd55df1d7f701d3cac4bba881c/lib/modules/dosomething/dosomething_reportback/includes/Reportback.php#L180-L185).